lib/models/event.dart

```diff
@@ -54,6 +54,13 @@ class Event {
         return EventType.Unknown;
     }
   }
+
+  @override
+  int get hashCode => name.hashCode + date.hashCode;
+
+  @override
+  bool operator ==(other) => this.hashCode == other.hashCode;
```

Comment:
If you want to write your own equality method you need to change this code !

The rules for equality (i.e. `hashCode` and `==`) are:  
1- Object that are equal must have the same `hashCode`
2- Objects that are not equal are allowed to have the same hash code
3- `hashCode` must only change if the object in a way that equality change
See the [documentation](https://api.dart.dev/stable/2.8.4/dart-core/Object/hashCode.html) for a complete definition of `hashCode` and `==`.

BTW these rules are available in all languages.

Also as the goal of `hashCode` is to provide an efficient way to retrieve an object in a collection (with a `O1` complexity), you have to avoid `hashCode` collision!. If you write
```dart
@override
  int get hashCode => 0
```
it works at it follows previous rules, but it's not efficient because all instances have the same `hashCode` so the complexity will be `On`: at worst you'll have to traverse all items in the collection to find a specific item. To minimize the collision, the most used algorithm is to shift each field:
```dart
int get hashCode {
    int hash = 7;
    hash = 31 * hash + (name.hashCode ?? 0);
    hash = 31 * hash + (date.hashCode ?? 0);
    return hash;
}
``` 

**But this first comment was for educational only** 

The purpose of the library `equatable` is exactly what you tried to do by hand. Just define properties which must be taken into account, and `equatable` overwrites `==` and `hashCode` for you in an efficient and safety way

Reply:
:+1: